### PR TITLE
Separate `Message`'s content and make messages immutable

### PR DESCRIPTION
**Pull request by limebell**
Separated `Message.Type` and `Message.DataField` into new class named `MessageContent`.

`Message` and `MessageContent` intended to be used immutable, thus all properties have only getter.

All messages (e.g `PingMsg`, `GetBlocksMsg`...) are now implements `MessageContent`. `ITransport` will wrap a `MessageContent` into a `Message`.

closes https://github.com/planetarium/libplanet/issues/2772

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -25,6 +49,9 @@ To be released.
 
  -  Added `Branch<T>` class to represent a sequence of consecutive `Block<T>`s.
     [[#2822]]
+ -  Added `MessageContent` class.  [[#2772], [#2831]]
+     -  All messages (eg. `PingMsg`) became to inherit `MessageContent`
```
Comment (greymistcube) [inline CHANGES.md:53]:
```suggestion
     -  All messages (e.g. `PingMsg`) became to inherit `MessageContent`
```

**Review by colibrishin — APPROVED** [review]

**Review by OnedgeLee — DISMISSED** [review]

File: CHANGES.md (hunk 1 above)
Comment (limebell) [reply CHANGES.md]: Amended! https://github.com/planetarium/libplanet/pull/2831/commits/af91bcdfab8b0b76d207af64044a2f83d5a42d05

**Review by colibrishin — APPROVED** [review]

File: Libplanet.Net/Protocols/KademliaProtocol.cs [hunk 2]
```diff
@@ -426,18 +426,14 @@ await PingAsync(found, _requestTimeout, cancellationToken)
                     timeout,
                     cancellationToken
                 ).ConfigureAwait(false);
-                if (!(reply is PongMsg pong))
+                if (!(reply.Content is PongMsg pong))
                 {
-                    throw new InvalidMessageException(
-                        $"Expected pong, but received {reply.Type}.", reply);
+                    throw new InvalidMessageContentException(
+                        "Expected pong, but received {reply.Content.Type}.", reply.Content);
```
Comment (greymistcube) [inline Libplanet.Net/Protocols/KademliaProtocol.cs:432]:
```suggestion
                        $"Expected pong, but received {reply.Content.Type}.", reply.Content);
```

Comment (limebell) [reply Libplanet.Net/Protocols/KademliaProtocol.cs]: Thanks, amended!

**Review by greymistcube — APPROVED** [review]

**Pull request merged**